File: utils/rename_img.py
import os

# Get the current working directory
folder_path = os.getcwd()

# Get all files in the directory
files = sorted(os.listdir(folder_path))
index = 0
# Initialize counter
for i, file_name in enumerate(files):
    # Get the full path of the file
    old_file_path = os.path.join(folder_path, file_name)
    
    # Get the file extension
    _, ext = os.path.splitext(file_name)

    # Check if the file is an image of the specified formats
    if ext.lower() in ['.bmp', '.jpg', '.jpeg', '.png', '.tif', '.tiff']:
        # Rename based on the index
        if index == 0:
            new_file_name = f"r{index:05d}{ext}"  # First image starts with 'r'
        else:
            new_file_name = f"d{index:05d}{ext}"  # Subsequent images start with 'd'

        new_file_path = os.path.join(folder_path, new_file_name)
        index = index + 1
        # Rename the file
        os.rename(old_file_path, new_file_path)

print("Renaming complete!")

File: utils/rename_img.m
% Get the current working directory
folder_path = pwd;

% Get all files in the directory
files = dir(fullfile(folder_path, '*.*'));
index = 1;
% Initialize counter
for i = 1:length(files)
    file_name = files(i).name;
    
    % Get the full path of the file
    old_file_path = fullfile(folder_path, file_name);
    
    % Get the file extension
    [~, ~, ext] = fileparts(file_name);

    % Check if the file is an image of the specified formats
    if ismember(lower(ext), {'.bmp', '.jpg', '.jpeg', '.png', '.tif', '.tiff'})
        % Rename based on the index
        if index == 1
            new_file_name = sprintf('r%05d%s', index - 1, ext);  % First image starts with 'r'
        else
            new_file_name = sprintf('d%05d%s', index - 1, ext);  % Subsequent images start with 'd'
        end
        index = index+1;
        new_file_path = fullfile(folder_path, new_file_name);
        
        % Rename the file
        movefile(old_file_path, new_file_path);
    end
end

disp('Renaming complete!');
